deps/rabbitmq_management/src/rabbit_mgmt_util.erl

```diff
@@ -210,6 +210,14 @@ vhost_from_headers(ReqData) ->
 vhost(ReqData) ->
     rabbit_web_dispatch_access_control:vhost(ReqData).
 
+vhost_or_bad_request(ReqData, Context) ->
+    case rabbit_web_dispatch_access_control:vhost(ReqData) of
+        not_found -> 
+            bad_request(rabbit_data_coercion:to_binary("vhost_not_found"),
```

Comment:
If a virtual host exists, shouldn't the most suitable HTTP return code be a 404 Not Found and not a 400 Bad Request?

Reply:
Yes, in those cases where both the vhost and the queue (or the vhost and a exchange, or a vhost and a binding) is the resource.

For instance, if a POST to `/api/queues/<vhost>/<queue>/get` failed with a 404 I would expect either the vhost or the queue for that host to be missing, because I understand the resource is the combination of both the vhost and the queue.

But on other more complex cases such as a POST to `/api/bindings/<vhost>/e/<src-exchange>/e/<dst-exchange>` a 404 would certainly not be enough. What's missing there? The source exchange on the current vhost, or the destination exchange on the current vhost? Or the vhost itself?

Regarding the proposed changes I limited myself to prevent 500 errors on posts, and since looking at previous code some cases were returning a 400, such as in [rabbit_mgmt_wm_definitions.erl](https://github.com/rabbitmq/rabbitmq-server/blob/dfddfe3fbb9070018b1be5d620f192f6d20b614b/deps/rabbitmq_management/src/rabbit_mgmt_wm_definitions.erl#L39), which solution is better, and coherent with the current design, wasn't clear.

I can let all changed callback modules fail with a 404 either using `resource_exists/2` and `allow_missing_post/2` or  `rabbit_mgmt_util:not_found`, which allows to specify the reason for the 404 (vhost_not_found); and revert `rabbit_mgmt_wm_definitions.erl` which already handled the missing vhost appropriately.